sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/CosmosRowConverter.scala

```diff
@@ -694,27 +689,86 @@ private[cosmos] class CosmosRowConverter(
   }
 
     private def parseTtlExpired(objectNode: ObjectNode): Boolean = {
-      getFullFidelityMetadata(objectNode) match {
-        case metadataNode: Some[ObjectNode] =>
-          metadataNode.get.get(TimeToLiveExpiredPropertyName) match {
-            case valueNode: JsonNode =>
-              Option(valueNode).fold(false)(v => v.asBoolean(false))
-            case _ => false
-          }
+      objectNode.get(MetadataJsonBodyAttributeName) match {
+        case metadataNode: JsonNode =>
+          metadataNode.get(TimeToLiveExpiredPropertyName) match {
+        case valueNode: JsonNode =>
+          Option(valueNode).fold(false)(v => v.asBoolean(false))
+        case _ => false
+        }
         case _ => false
       }
     }
 
-    private def parseOperationType(objectNode: ObjectNode): String = {
-      getFullFidelityMetadata(objectNode) match {
-        case metadataNode: Some[ObjectNode] =>
-          metadataNode.get.get(OperationTypePropertyName) match {
+    private def parseId(objectNode: ObjectNode): String = {
+        val currentNode = getCurrentOrPreviousNode(objectNode)
+        currentNode.get(IdAttributeName) match {
             case valueNode: JsonNode =>
-              Option(valueNode).fold(null: String)(v => v.asText(null))
+                Option(valueNode).fold(null: String)(v => v.asText(null))
             case _ => null
-          }
-        case _ => null
+        }
+    }
+
+    private def parseETag(objectNode: ObjectNode): String = {
+        val currentNode = getCurrentOrPreviousNode(objectNode)
+        currentNode.get(ETagAttributeName) match {
+            case valueNode: JsonNode =>
+                Option(valueNode).fold(null: String)(v => v.asText(null))
+            case _ => null
+        }
+    }
+
+    private def getCurrentOrPreviousNode(objectNode: ObjectNode): JsonNode = {
+      var currentNode = objectNode.get(CurrentAttributeName)
+      if (currentNode == null || currentNode.isEmpty) {
+        currentNode = objectNode.get(PreviousRawJsonBodyAttributeName)
       }
+      currentNode
+    }
+
+    //  Timestamp always returns the crts (conflict resolution timestamp).
+    //  For single-master, crts will always be same as _ts
+    //  For multi-master, crts will be the latest resolution timestamp of any conflicts
+    private def parseTimestamp(objectNode: ObjectNode): Long = {
+        objectNode.get(MetadataJsonBodyAttributeName) match {
+            case metadataNode: JsonNode =>
```

Comment:
what if objectNode does not have MetadataJsonBodyAttribute

Reply:
it will always have the metadata json body attribute name. 
private[spark] val defaultFullFidelityChangeFeedSchemaForInferenceDisabled = StructType(Seq(
    StructField(RawJsonBodyAttributeName, StringType, nullable=true),
    StructField(IdAttributeName, StringType, nullable=false),
    StructField(TimestampAttributeName, LongType, nullable=false),
    StructField(ETagAttributeName, StringType, nullable=false),
    StructField(LsnAttributeName, LongType, nullable=false),
    StructField(MetadataJsonBodyAttributeName, StringType, nullable=false),
    StructField(PreviousRawJsonBodyAttributeName, StringType, nullable=true),
    StructField(OperationTypeAttributeName, StringType, nullable=false),
    StructField(CrtsAttributeName, LongType, nullable=false),
    StructField(PreviousImageLsnAttributeName, LongType, nullable=true)
  ))

It is not nullable.